api/src/models/user/user.methods.ts

```diff
@@ -198,22 +198,48 @@ export async function createAndPostThread(this: IUserDocument, threadDetails: IT
     },
     comments: {},
     likes: {},
-    shares: {}
+    shares: {},
+    createdAt: new Date(),
+    updatedAt: new Date()
   };
-  try {
-    const newlyCreatedThread = await ThreadModel.create(userThread);
-    this["threads"]["started"][`${newlyCreatedThread.id.toString()}`] = newlyCreatedThread;
-    // Forces the parent object to update: there may be a better way
-    this.markModified("threads");
-    await this.save();
-    return {userData: this, threadData: newlyCreatedThread};
-  } catch (err) {
-    console.log(err);
-  }
+
+  const newlyCreatedThread = await ThreadModel.create(userThread);
+  this["threads"]["started"][`${newlyCreatedThread.id.toString()}`] = newlyCreatedThread;
+  // Forces the parent object to update: there may be a better way
+  this.markModified("threads");
+  await this.save();
+  return {userData: this, threadData: newlyCreatedThread};
+
 }
 
-export async function deleteUserThreadByThreadId(this: IUserDocument, threadId: string) {
+/** This is a modular helper method. This will only
+ * return a sorted list (by date latest) of threads from source user's connections
+ */
+export async function getConnectionThreads(this: IUserDocument): Promise<Array<IThread>> {
+  // Get an array of userIds for this.connections
+  const connectionUserIds = Object.keys(this.connections);
+
+  // Find user documents that match the ids in the above array
+  const users = await UserModel.find().where("_id").in(connectionUserIds).exec();
+  const threads: IThread[] = [];
+
+  users.forEach((user) => {
+    for (const[_, value] of Object.entries(user.threads.started)) {
+      threads.push(value);
+    }
+  });
 
+  // Sort the threads and return
+  return threads.sort((a, b) => {
```

Comment:
This could be simplified 
return threads.sort((a, b) => b.createdAt.valueOf() - a.createdAt.valueOf());

Reply:
I was testing out sorting dates in the node SLI and it wasn't sorting properly with this syntax. But I updated it and tests are passing, so good!